### PR TITLE
[DO NOT MERGE] Adding InstanceFailoverGroups 

### PR DESCRIPTION
**Pull request by jugeorge**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by jhendrixMSFT** [comment]
@AutorestCI rebuild azure-sdk-for-go

**Comment by jhendrixMSFT** [comment]
@AutorestCI rebuild azure-sdk-for-go

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/instanceFailoverGroups.json [hunk 1]
```diff
@@ -0,0 +1,644 @@
...
+    "InstanceFailoverGroup": {
+      "description": "An instance failover group.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
```
Comment (alvadb) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/instanceFailoverGroups.json]: Does it need to have properties within properties?

**Review by alvadb — CHANGES_REQUESTED** [review]

**Review by alvadb — APPROVED** [review]

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/instanceFailoverGroups.json (hunk 1 above)
Comment (jaredmoo) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/instanceFailoverGroups.json]: Isn't this right? Outer "properties" is swagger syntax, inner "properties" is the name of the actual json element.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/instanceFailoverGroups.json [hunk 2]
```diff
@@ -0,0 +1,644 @@
...
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "InstanceFailoverGroupReplicationRole",
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "ManagedInstancePairInfo": {
```
Comment (jaredmoo) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/instanceFailoverGroups.json]: Can this be changed to just ManagedInstancePair ? "Info" seems redundant. All objects are information about something.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/instanceFailoverGroups.json [hunk 3]
```diff
@@ -0,0 +1,644 @@
...
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "ReadOnlyEndpointFailoverPolicy",
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "PartnerRegionInfo": {
```
Comment (jaredmoo) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/instanceFailoverGroups.json]: same comment about "Info" here

**Comment by lmazuel** [comment]
@AutorestCI regenerate azure-sdk-for-python

**Pull request merged**